embrace-android-features/src/test/java/io/embrace/android/embracesdk/internal/capture/activity/UiLoadTraceEmitterTest.kt

```diff
@@ -0,0 +1,427 @@
+package io.embrace.android.embracesdk.internal.capture.activity
+
+import android.os.Build
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.embrace.android.embracesdk.assertions.assertEmbraceSpanData
+import io.embrace.android.embracesdk.fakes.FakeClock
+import io.embrace.android.embracesdk.fakes.injection.FakeInitModule
+import io.embrace.android.embracesdk.internal.payload.toNewPayload
+import io.embrace.android.embracesdk.internal.spans.SpanService
+import io.embrace.android.embracesdk.internal.spans.SpanSink
+import io.embrace.android.embracesdk.internal.utils.BuildVersionChecker
+import io.opentelemetry.api.trace.SpanId
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import kotlin.math.max
+import kotlin.math.min
+
+@RunWith(AndroidJUnit4::class)
+internal class UiLoadTraceEmitterTest {
+    private lateinit var clock: FakeClock
+    private lateinit var spanSink: SpanSink
+    private lateinit var spanService: SpanService
+    private lateinit var traceEmitter: UiLoadTraceEmitter
+
+    @Before
+    fun setUp() {
+        clock = FakeClock()
+        val initModule = FakeInitModule(clock = clock)
+        spanSink = initModule.openTelemetryModule.spanSink
+        spanService = initModule.openTelemetryModule.spanService
+        spanService.initializeService(clock.now())
+        clock.tick(100L)
+        traceEmitter = UiLoadTraceEmitter(
+            spanService = spanService,
+            versionChecker = BuildVersionChecker,
+        )
+    }
+
+    @Config(sdk = [Build.VERSION_CODES.UPSIDE_DOWN_CAKE])
+    @Test
+    fun `verify cold open trace from another activity in U`() {
+        verifyOpen(
+            previousState = PreviousState.FROM_ACTIVITY,
+            uiLoadType = UiLoadType.COLD,
+            firePreAndPost = true,
+            hasRenderEvent = true,
+        )
+    }
+
+    @Config(sdk = [Build.VERSION_CODES.UPSIDE_DOWN_CAKE])
+    @Test
+    fun `verify cold open trace from the same activity in U`() {
+        verifyOpen(
+            lastActivityName = ACTIVITY_NAME,
+            previousState = PreviousState.FROM_ACTIVITY,
+            uiLoadType = UiLoadType.COLD,
+            firePreAndPost = true,
+            hasRenderEvent = true,
+        )
+    }
+
+    @Ignore("Not working yet")
```

Comment:
Test cases with the ignore annotation should be updated to run before merging, or can be removed if they're no longer relevant

Reply:
Ah, I think they work - so I'll either un-ignore or fix them